src/test/java/net/openhft/chronicle/threads/ThreadsTestCommon.java

```diff
@@ -59,8 +61,22 @@ public void checkExceptions() {
         if (Jvm.hasException(exceptions)) {
             Jvm.dumpException(exceptions);
             Jvm.resetExceptionHandlers();
-            Assert.fail();
+            fail();
+        }
+    }
+
+    public void assertExceptionThrown(String message) {
+        String description = format("No exception found containing string `%s`", message);
+        assertExceptionThrown(k -> k.message.contains(message) || (k.throwable != null && k.throwable.getMessage().contains(message)), description);
+    }
+
+    public void assertExceptionThrown(Predicate<ExceptionKey> predicate, String description) {
+        for (ExceptionKey key : exceptions.keySet()) {
+            if (predicate.test(key)) {
+                return;
+            }
         }
+        fail(description);
```

Comment:
expectException doesn't fail the test, this one is to ensure the warning was logged